src/crimson/os/seastore/seastore_types.cc

```diff
@@ -94,22 +94,379 @@ std::ostream& operator<<(std::ostream& out, segment_seq_printer_t seq)
+    out << "clone_prefix";
+    break;
+  case laddr_conflict_condition_t::all:
+    out << "all";
+    break;
+  default:
+    __builtin_unreachable();
+  }
+  out << ", conflict_policy=";
+  switch(hint.policy) {
+  case laddr_conflict_policy_t::gen_random:
+    out << "gen_random";
+    break;
+  case laddr_conflict_policy_t::linear_search:
+    out << "linear_search";
+    break;
+  case laddr_conflict_policy_t::never:
+    out << "never";
+    break;
+  default:
+    __builtin_unreachable();
+  }
+  return out << ", block_size=" << hint.block_size << ")";
+}
+
+namespace {
+struct random_generator_t {
+  static random_generator_t &get() {
+    static thread_local random_generator_t r{};
+    return r;
+  }
+  random_generator_t()
+      : eng(std::rand()),
+        global(0, std::numeric_limits<uint64_t>::max()) {}
+  std::default_random_engine eng;
+  std::uniform_int_distribution<uint64_t> global;
+  uint64_t operator()() { return global(eng); }
+};
+uint64_t rand_field() { return random_generator_t::get()(); }
+extent_len_t get_block_size_mask(extent_len_t block_size) {
+  assert(block_size != 0 && (block_size & (block_size - 1)) == 0);
+  return (block_size >> laddr_t::UNIT_SHIFT) - 1;
+}
+} // namespace
+
+laddr_hint_t laddr_hint_t::create_global_md_hint() {
+  laddr_t addr = L_ADDR_MIN.with_object_content(rand_field());
+  assert(addr.get_object_info() == 0);
+  return {
+    addr,
+    laddr_conflict_condition_t::object_content,
+    laddr_conflict_policy_t::linear_search,
+    /*block_size=*/ laddr_t::UNIT_SIZE
+  };
+}
+
+laddr_hint_t laddr_hint_t::create_onode_hint(
+  shard_t shard,
+  pool_t pool,
+  crush_hash_t crush)
+{
+  laddr_t addr = L_ADDR_MIN;
+  addr.set_shard(shard);
+  addr.set_pool(pool);
+  addr.set_reversed_hash(crush);
+
+  assert(addr.match_shard_bits(shard));
+  assert(addr.match_pool_bits(pool));
+  assert(addr.get_reversed_hash() == crush);
+  assert(addr.get_local_object_id() == 0);
+  assert(addr.get_object_content() == 0);
+
+  return {
+    addr,
+    laddr_conflict_condition_t::object_content,
+    laddr_conflict_policy_t::linear_search,
+    /*block_size=*/ laddr_t::UNIT_SIZE
+  };
+}
+
+laddr_hint_t laddr_hint_t::create_fresh_object_data_hint(
+  shard_t shard,
+  pool_t pool,
+  crush_hash_t crush)
+{
+  auto addr = L_ADDR_MIN;
+  auto condition = laddr_conflict_condition_t::object_prefix;
+  auto policy = laddr_conflict_policy_t::gen_random;
+
+  addr.set_shard(shard);
+  addr.set_pool(pool);
+  addr.set_reversed_hash(crush);
+
+  do {
+    addr.set_local_object_id(rand_field());
+    addr.set_local_clone_id(rand_field());
+  } while (!addr.is_object_address());
+
+  assert(addr.match_pool_bits(pool));
+  assert(addr.match_shard_bits(shard));
+  assert(addr.get_reversed_hash() == crush);
+  assert(addr.get_local_object_id() != LOCAL_OBJECT_ID_ZERO);
+  assert(!addr.is_metadata());
+  assert(addr.get_offset_bytes() == 0);
+  return {addr, condition, policy, /*block_size=*/ laddr_t::UNIT_SIZE};
```

Comment:
Probably 258~269 above can be replaced by `return addr.find_next_random()` to generate random, and the asserts are unnecessary if duplicated in `addr.find_next_random()`?

Same to all the other `laddr_hint_::create_*()`.